### PR TITLE
chore: Improve overall performance

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
#295 added login functionality. This, however, introduced an unnecessary re-render of the whole app. This PR removes the need for this re-render, thus reducing TBT and increasing overall performance

## How does it work?
#295 made the return type of the `person` query to be `StorePerson!`. This means that the query result of person returns the following value for the anonymous user:
```json
{
  "id": "",
  "email": "",
  "givenName": "",
  "familyName": ""
}
```

This, in turn, is different from the default user session, passed in the [context provider](https://github.com/vtex-sites/base.store/blob/0a16a3784749af1f60bc359d7a5cbafb8b9783ef/gatsby-browser.js#L29) as `null`. This mismatch in values, caused the session to be reset for the anonymous user, causing an unnecessary re-render.

To fix this issue, I made the api return `null` for the `person` query when executed by an anonymous user. This matches the default value from the session, this not triggering a re-render. All changes are performed on this [PR](https://github.com/vtex/faststore/pull/1190) at faststore

An evidence can be seen below were the first image shows the `ProductGallery` component being rendered twice (due to session being reset) and the second image shows the absence of this re-render after the proposed fix. 

image
image

## How to test it?
Make sure everything works as before and `ProductGallery` renders once. 
> Make sure to delete any pre-stored value in indexedDB

## Checklist
- [x] CHANGELOG entry added

### CODE REVIEW

**Review by emersonlaurentino — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `main::store::postalCode` value from storage.
 
 ### Fixed
-
+- Unecessary app rerender after login feature
```
Comment (lariciamota) [inline CHANGELOG.md:69]:
```suggestion
- Unnecessary app rerender after login feature
```

**Comment by igorbrasileiro** [comment]
Hey, you changed nothing. The purpose is to add just the mark hoc? @tlgimenes

File: src/components/sections/ProductGallery/ProductGallery.tsx [hunk 2]
```diff
@@ -173,4 +174,5 @@ function ProductGallery({ title, searchTerm }: Props) {
   )
 }
 
-export default ProductGallery
+ProductGallery.displayName = 'ProductGallery'
```
Comment (filipewl) [inline src/components/sections/ProductGallery/ProductGallery.tsx:177]: Is it worth adding a test in `performance.test.js` to assert it's only rendered once for the PLP?

**Review by filipewl — COMMENTED** [review]
Following @igorbrasileiro's [question](https://github.com/vtex-sites/base.store/pull/418#issuecomment-1078209354), I assume this PR is missing the part where the [faststore](https://github.com/vtex/faststore/pull/1190) package is updated to reflect the improvements.

**Comment by tlgimenes** [comment]
> Following @igorbrasileiro's [question](https://github.com/vtex-sites/base.store/pull/418#issuecomment-1078209354), I assume this PR is missing the part where the [faststore](https://github.com/vtex/faststore/pull/1190) package is updated to reflect the improvements.

Yes, this part was missing, I had only done it locally for extracting the images. Now we have the full version

File: src/components/sections/ProductGallery/ProductGallery.tsx (hunk 2 above)
Comment (tlgimenes) [reply src/components/sections/ProductGallery/ProductGallery.tsx:177]: Nice!

**Comment by igorbrasileiro** [comment]
Results after run 20 psi tests via api
Page result - https://base.vtex.app/
| Metric | Mean | Standard deviation |
|--------|--------|--------|
| Cumulative Layout shift (CLS) | 0.00 | 0.00 |
| First Contentful Paint (FCP) | 1823.92 | 72588.98 |
| First Contentful Paint 3g (FCP) | 3398.24 | 159152.47 |
| Largest Contentful Paint (LCP) | 1918.23 | 11251.40 |
| Time to Interactive (TTI) | 4095.18 | 218320.64 |
| Total Blocking Time (TBT) | 722.31 | 66249.29 |
| Performance score | 0.802 | 0.005419 |
| JavaScript Execution Time | 2221.49 | 349446.83 |
| Speed Index | 2512.07 | 66531.67 |

Page result - https://sfj-3321c99--base.preview.vtex.app/
| Metric | Mean | Standard deviation |
|--------|--------|--------|
| Cumulative Layout shift (CLS) | 0.00 | 0.00 |
| First Contentful Paint (FCP) | 1408.29 | 4461.97 |
| First Contentful Paint 3g (FCP) | 2592.67 | 74760.67 |
| Largest Contentful Paint (LCP) | 1935.69 | 4314.01 |
| Time to Interactive (TTI) | 3523.45 | 0.01 |
| Total Blocking Time (TBT) | 515.45 | 7805.14 |
| Performance score | 0.855 | 0.000525 |
| JavaScript Execution Time | 1685.06 | 27049.33 |
| Speed Index | 2034.19 | 10235.46 |

**Review by igorbrasileiro — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
     "@envelop/graphql-jit": "^1.1.1",
     "@envelop/parser-cache": "^2.2.0",
     "@envelop/validation-cache": "^2.2.0",
-    "@faststore/api": "^1.6.16",
+    "@faststore/api": "https://pkg.csb.dev/vtex/faststore/commit/60cdb1f2/@faststore/api",
```
Comment (lariciamota) [inline package.json:30]: Just a reminder to update it before merging

**Review by lariciamota — APPROVED** [review]
Nice!! 🚀

**Pull request merged**